File: 14_이정형/10828.py
# -*- coding: utf-8 -*-
import sys

stack = []


def push(x):
    stack.append(x)


def pop():
    if not stack:
        return -1
    else:
        return stack.pop()


def size():
    return len(stack)


def top():
    if stack:
        return stack[-1]
    else:
        return -1


def empty():
    if stack:
        return 0
    else:
        return 1


# 총 몇개를 받을 건지
num = int(sys.stdin.readline())
# 함수 적용
for _ in range(num):
    input_split = sys.stdin.readline().split()
    order = input_split[0]

    if order == "push":
        push(input_split[1])
    elif order == "pop":
        print(pop())
    elif order == "size":
        print(size())
    elif order == "empty":
        print(empty())
    elif order == "top":
        print(top())

File: 24_김수빈/10828.c
// 작성일: 2021년 11월 14일
// 작성일: 2022년 1월 17일
// 작성자: 김수빈
// 프로그램명: 스택
#include <stdio.h>
#include <string.h>
int stack[10000];
int size = 0;

int empty() {
	if (size == 0) return 1;
	return 0;
}
void push(int data) {
	stack[size++] = data;
}
int pop() {
	if (empty()) return -1;
	return stack[--size];
}
int top() {
	if (empty()) return -1;
	return stack[size - 1];
}

int main() {
	int num, data;
	char str[6] = { 0 };

	scanf("%d", &num);
	for (int i = 0; i < num; i++) {
		scanf("%s", str);

		if (!strcmp(str, "push")) {
			scanf("%d", &data);
			push(data);
		}
		else if (!strcmp(str, "pop"))
			printf("%d\n", pop());
		else if (!strcmp(str, "size"))
			printf("%d\n", size);
		else if (!strcmp(str, "top"))
			printf("%d\n", top());
		else if (!strcmp(str, "empty"))
			printf("%d\n", empty());
	}
	return 0;
}